### DISCUSSION TITLE
Error while using connectOrCreate in a nested create

### DISCUSSION THREAD

**Discussion by AtiArabshahi** [prisma/prisma#18977] [Q&A]
Hello,

I'm using GraphQL and Prisma to create a backend app. I have Publication, Version and Author models defined in Prisma. Each Publication can have many versions and each version belongs to one publication. Also there's a many-to-many relation between Version and Author. I'm trying to create the createPublication mutation as follow:

```
const { accessType, isPublished, tags, version } = args.input;

const publication = await prisma.publication.create({
  data: {
    accessType,
    isPublished,
    tags,
    createdAt: new Date(),
    updatedAt: new Date(),
    versions: {
      create: [{
        versionNumber: version.versionNumber,
        title: version.title,
        abstract: version.abstract,
        authors: {
          connectOrCreate: version.authors.map((author) => {
            return {
              where: {
                firstName: author.firstName,
                middleName: author.middleName || null, // Since the middleName is option, I have set it to be null if not exsists in `where` clause.
                lastName: author.lastname,
              },
              create: {
                firstName: author.firstName,
                middleName: author.middleName,
                lastName: author.lastname,
              },
            }
          })
        },
      }],
    },
  },
)};
```

The Author model in Prisma is like:
```
model Author {
  id         String  @id @default(uuid())
  userId     String? @unique
  firstName  String
  middleName String?
  lastName   String

  user         User?                @relation(fields: [userId], references: [id])
  versions     Version[]
}
```

The user field in Author model is for when an author is already a user of this app.

Running this code, I get the following error:
```
Types of property 'where' are incompatible.
Type '{ firstName: string; middleName: string; lastName: string; }' has no properties in 
common with type 'AuthorWhereUniqueInput'.
```

I can't figure out what is the problem. I'll appreciate any help.

**Comment by ludralph** [comment]
Hi @AtiArbshhi 👋 

Can you please enable the preview feature `extendedWhereUnique` as mentioned in this section of the [documentation](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#filter-on-non-unique-fields-with-userwhereuniqueinput)? 

Once this has been done, retry the query and let me know if you still have issues.

**Reply by ludralph** [reply]
Hi @AtiArbshhi 

Do you have any more questions about this issue? If not, it would be really helpful if you could quickly mark this Discussion as answered to indicate that your issue has been resolved.

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Reply by ludralph** [reply]
Hi @AtiArbshhi 

do you have any more questions about this issue? If not, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Reply by ludralph** [reply]
The user seems to have figured out a solution themselves and is not following up with us anymore. If you have the same or similar problem, please open a new discussion. Thank you!
